package NSaasTrie;

option optimize_for = SPEED;

message TRealm {
    optional string Name = 1;
    repeated string Key = 2;
    optional uint32 Shards = 3; // obsolete
}

message TComplexKey {
    optional uint64 KeyPrefix = 1;
    optional string MainKey = 2;
    repeated string KeyRealms = 3;
    repeated TRealm AllRealms = 4;

    optional bool LastRealmUnique = 5; // UserRegion

    optional uint64 RealmPrefixSet = 6;

    optional string UrlMaskPrefix = 7;
}
